test/common/http/header_map_impl_test.cc

```diff
@@ -319,112 +319,54 @@ TEST(HeaderMapImplTest, DoubleInlineAdd) {
   EXPECT_EQ(1UL, headers.size());
 }
 
-struct HeaderBrutalityResult {
-  const HeaderString& val;
-  const void* keyPtr;
-};
-
-const HeaderBrutalityResult doHeaderBrutality(HeaderMapImpl& headers, LowerCaseString& lcKey) {
-  HeaderBrutalityResult result{headers.get(lcKey)->value(),
-                               static_cast<const void*>(lcKey.get().data())};
-
-  // This is... not exactly legal, but let's do it anyway. It should
-  // safe, since lcKey is still in scope here.
-  memcpy(const_cast<void*>(result.keyPtr), static_cast<const void*>("XXXXX"), 5);
-
-  // Make sure our brutality worked.
-  EXPECT_STREQ("XXXXX", lcKey.get().c_str());
-
-  return result;
-}
-
-const HeaderBrutalityResult addAHeaderString(HeaderMapImpl& headers) {
-  // This needs to not be a parameter. Part of the test is that the
-  // key we're using actually goes out of scope before we check the
-  // HeaderMap for the value we need.
-  LowerCaseString lcKey("hello");
-
-  headers.addCopy(lcKey, "world");
-
-  return doHeaderBrutality(headers, lcKey);
-}
-
-const HeaderBrutalityResult addAHeaderInt(HeaderMapImpl& headers) {
-  // This needs to not be a parameter. Part of the test is that the
-  // key we're using actually goes out of scope before we check the
-  // HeaderMap for the value we need.
-  LowerCaseString lcKey("hello");
-
-  headers.addCopy(lcKey, 42);
-
-  return doHeaderBrutality(headers, lcKey);
-}
-
-// void dumpHeaders(HeaderMapImpl& headers, int idx) {
-//   std::cerr << "---- dumpHeaders " << idx << std::endl;
-
-//   headers.iterate(
-//     [](const HeaderEntry& header, void* ctx) -> void {
-//       (void)ctx;  // Shut up the "unused parameter" error
-
-//       std::string key(header.key().c_str());
-//       std::string value(header.value().c_str());
-
-//       std::cerr << key << ": "
-//                 << value
-//                 << std::endl;
-//     },
-//     nullptr
-//   );
-// }
-
 TEST(HeaderMapImplTest, AddCopy) {
   HeaderMapImpl headers;
 
   // Start with a string value.
-  HeaderBrutalityResult v1 = addAHeaderString(headers);
-  // dumpHeaders(headers, 1);
-
-  EXPECT_STREQ("world", v1.val.c_str());
-  EXPECT_EQ(5UL, v1.val.size());
-  EXPECT_EQ(1UL, headers.size());
+  LowerCaseString* lcKeyPtr(new LowerCaseString("hello"));
```

Comment:
I think it's still better to use unique_ptr here.  It's less important for the test but it's a good habit to use the smart pointer types consistently.

Use `std::unique_ptr<LowerCaseString> lcKeyPtr(new LowerCaseString("hello")` here, and then where you call delete below use `lcKeyPtr.reset()` which will implicitly call delete.